backends/tfhe-cuda-backend/src/cuda_bind.rs

```diff
@@ -3,104 +3,58 @@ use std::ffi::c_void;
 #[link(name = "tfhe_cuda_backend", kind = "static")]
 extern "C" {
```

Comment:
"stupid" question are we sure the C ABI is respected when the CUDA backend is compiled ? in the sense that C++ does not exactly have the C ABI and I don't know how both relate 🤔

Reply:
(related to the potential launch failure stuff)

Reply:
Hmm I'm not totally sure no

Reply:
this maybe https://isocpp.org/wiki/faq/mixing-c-and-cpp#call-c

Reply:
this way you make sure in the C++ code base that you expose the C function as extern "C"

Reply:
We should be alright, as all the entry points in the binding are actually "C compatible". Nothing C++-like is passed through the interface, only raw pointer types.

Reply:
while I agree, I would not be against the "ceinture et bretelles" approach here :)

Reply:
And we define all functions that go through the interface with "extern C" in the header files.

Reply:
so then we are good on the ABI declaration, nothing else to do on the C++ side, just need to make sure definitions agree, do you want someone to double check function by function ?

Reply:
I think this would be very troublesome and not so efficient: I already went through it thoroughly yesterday. But really at some point we should spend some time to figure out how to generate this API automatically, at least part of it.